Google.Api.Gax/Platform.cs

```diff
@@ -43,6 +43,116 @@ public enum PlatformType
         /// Execution platform is Google Container Engine (Kubernetes).
         /// </summary>
         Gke = 3,
+
+        /// <summary>
+        /// Execution platform is Google Cloud Run.
+        /// </summary>
+        CloudRun = 4
+    }
+
+    /// <summary>
+    /// Google Cloud Run details.
+    /// </summary>
+    public sealed class CloudRunPlatformDetails
+    {
+        /// <summary>
+        /// Builds a <see cref="CloudRunPlatformDetails"/> from the given metadata
+        /// and Cloud Run environment variables.
+        /// The metadata is normally retrieved from the GCE metadata server.
+        /// </summary>
+        /// <param name="metadataJson">JSON metadata, normally retrieved from the GCE metadata server.
+        /// Must not be <c>null</c>.</param>
+        /// <returns>A populated <see cref="CloudRunPlatformDetails"/> if the metadata represents and GCE instance;
+        /// <c>null</c> otherwise.</returns>
+        public static CloudRunPlatformDetails TryLoad(string metadataJson)
+        {
+            GaxPreconditions.CheckNotNull(metadataJson, nameof(metadataJson));
+            JObject metadata;
+            try
+            {
+                metadata = JObject.Parse(metadataJson);
+            }
+            catch
+            {
+                return null;
+            }
+            
+            var projectId = metadata["project"]?["projectId"]?.ToString();
+            var zoneName = metadata["instance"]?["zone"]?.ToString();
+            if (projectId == null || zoneName == null || !Platform.s_zoneTemplate.TryParseName(zoneName, out var zoneResourceName))
+            {
+                return null;
+            }
+            string location = zoneResourceName[1];
+            // FIXME: Temporary hack, do not commit
```

Comment:
Should this be fixed?

Reply:
That's weird - I thought I'd pushed a version which had that removed. Will check.

Reply:
Doh, hadn't committed the change. Removed.